### PR TITLE
Introduces a simple dialog to merge assets

### PR DESCRIPTION
**Pull request by kelsos**

### CODE REVIEW

File: frontend/app/src/components/asset-manager/MergeDialog.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    this.pending = true;
+    const { success, message } = await this.mergeAssets({
+      sourceIdentifier: this.source,
+      targetIdentifier: this.target
+    });
+
+    if (success) {
+      this.input(false);
+    } else {
+      this.errorMessages.push(message ?? this.$t('merge_dialog.error').toString())
```
Comment (LefterisJP) [inline frontend/app/src/components/asset-manager/MergeDialog.vue:103]: Github CI seems to have hit something here.

File: frontend/app/src/locales/en.json [hunk 2]
```diff
@@ -350,6 +350,23 @@
     "terminate": "Terminate",
     "message": "rotki's minimum supported macOs version is Mojave(10.14)"
   },
+  "merge_dialog": {
+    "cancel": "Cancel",
+    "error": "Merge was not successful",
+    "hint": "The source asset will be merged into the target assets and all it's usages will be updated.",
```
Comment (LefterisJP) [inline frontend/app/src/locales/en.json:356]:
```suggestion
    "hint": "The source asset will be merged into the target assets and all its usages will be updated.",
```

File: frontend/app/src/locales/en.json [hunk 3]
```diff
@@ -350,6 +350,23 @@
...
+    "merge": "Merge",
+    "source": {
+      "label": "Source",
+      "non_empty": "The source identifier cannot not be empty"
+    },
+    "source_hint": "The asset identifier that wou want to merge into the target",
```
Comment (LefterisJP) [inline frontend/app/src/locales/en.json:362]:
```suggestion
    "source_hint": "The asset identifier that you want to merge into the target",
```

**Review by LefterisJP — COMMENTED** [review]
Some small comments. Did no try it yet. How can I see the identifier of existing assets? We said we also need this.

**Comment by LefterisJP** [comment]
Also just made a rebase on develop

**Review by LefterisJP — COMMENTED** [review]
At merge success remember to refresh everywhere where the source asset was used.

File: frontend/app/src/components/asset-manager/MergeDialog.vue (hunk 1 above)
Comment (LefterisJP) [reply frontend/app/src/components/asset-manager/MergeDialog.vue]: It may have gone away after running the linter.

Comment (LefterisJP) [reply frontend/app/src/components/asset-manager/MergeDialog.vue]: Done.

File: frontend/app/src/locales/en.json (hunk 3 above)
Comment (LefterisJP) [reply frontend/app/src/locales/en.json:362]: done.

**Review by LefterisJP — APPROVED** [review]
Rebased on develop, tested it and fixed the aforementioned typos.

The two things remaining to be done as discussed are:

1. Add an easier way to copy the ID of an asset
2. Refresh all places where the replaced asset may have been in the frontend.

**Comment by kelsos** [comment]
> Rebased on develop, tested it and fixed the aforementioned typos.
> 
> The two things remaining to be done as discussed are:
> 
>     1. Add an easier way to copy the ID of an asset
> 
>     2. Refresh all places where the replaced asset may have been in the frontend.

For the second part are we doing it as we discussed by showing a notification to the user to let them know that they will have to manual refresh?

Otherwise, I would either have to refresh everything or traverse through the whole frontend state and figure out where the asset might be used. (this would be every event, trade, ledger action, balance, etc which would be a bit complicated)

**Comment by LefterisJP** [comment]
> Otherwise, I would either have to refresh everything or traverse through the whole frontend state and figure out where the asset might be used. (this would be every event, trade, ledger action, balance, etc which would be a bit complicated)

Let's start the easy way and just tell the user they would have to either refresh each place the asset was used or more simply restart theapp.

File: frontend/app/src/locales/en.json [hunk 4]
```diff
@@ -350,6 +353,25 @@
     "terminate": "Terminate",
     "message": "rotki's minimum supported macOs version is Mojave(10.14)"
   },
+  "merge_dialog": {
+    "cancel": "Cancel",
+    "close": "Close",
+    "done": "The asset merge is complete. The old asset will still appear in the frontend until the next manual refresh or re-login.",
```
Comment (LefterisJP) [inline frontend/app/src/locales/en.json:359]: This sounds a bit like no action is needed. Perhaps use imperative form.
```suggestion
    "done": "The asset merge is complete. The old asset will still appear in the frontend. To make sure the new one is shown everywhere either restart the application or manually refresh balances in each place the old asset appears.",
```

**Review by LefterisJP — APPROVED** [review]
Works. Small comment on the merge success popup.

**Pull request merged**